brage-import/src/test/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumerTest.java

```diff
@@ -1023,6 +1025,44 @@ void shouldMergePublicationContextAndUseCourseFromBragePublicationWhenPublicatio
         assertThat(((UnconfirmedCourse) actualPublicationContext.getCourse()).code(), is(equalTo(subjectCode)));
     }
 
+    @Test
+    void shouldKeepOrderOfRelatedDocumentsWhenMergingDegreePhd()
+        throws IOException, InvalidUnconfirmedSeriesException {
+        var cristinIdentifier = randomString();
+        var publication = randomPublication(DegreePhd.class);
+        publication.setAdditionalIdentifiers(Set.of());
+        publication.setAdditionalIdentifiers(Set.of(cristinAdditionalIdentifier(cristinIdentifier)));
+        publication.getEntityDescription().getReference().setDoi(null);
+        publication.getEntityDescription().getReference().setPublicationContext(new Degree(null, null, null, null,
+                                                                                           List.of(), null));
+        publication.getEntityDescription().setPublicationDate(new no.unit.nva.model.PublicationDate.Builder().withYear("2022").build());
+        publication.getEntityDescription().setMainTitle("Dynamic - Response of Floating Wind Turbines! Report");
+        var existingPublication = resourceService.createPublicationFromImportedEntry(publication);
+        var contributor = existingPublication.getEntityDescription().getContributors().getFirst();
+        var brageContributor = new Contributor(new Identity(contributor.getIdentity().getName(), null),
+                                               "ARTIST", null, List.of());
+        var subjectCode = randomString();
+        var brageGenerator = new NvaBrageMigrationDataGenerator.Builder().withType(TYPE_PHD)
+                                 .withCristinIdentifier(cristinIdentifier)
+                                 .withMainTitle("Dynamic - Response of Floating Wind Turbines! Report")
+                                 .withContributor(brageContributor)
+                                 .withHasPart(List.of("1", "2"))
+                                 .withPublicationDate(new PublicationDate("2023",
+                                                                          new PublicationDateNva.Builder().withYear("2023").build()))
+                                 .withSubjectCode(subjectCode)
+                                 .build();
+        var s3Event = createNewBrageRecordEvent(brageGenerator.getBrageRecord());
+        var actualPublication = handler.handleRequest(s3Event, CONTEXT);
+
+        var actualPublicationContext = ((DegreePhd) actualPublication.publication().getEntityDescription()
+                                                     .getReference()
+                                                     .getPublicationInstance());
+
+        var related = actualPublicationContext.getRelated().stream().toList();
+        assertThat(related.getLast(), is(equalTo(new UnconfirmedDocument("2"))));
+        assertThat(related.get(related.size() - 2), is(equalTo(new UnconfirmedDocument("1"))));
+    }
```

Comment:
Ugly assertion, but have no idea how to do other way when handling a set.

Reply:
Could hasItem plus assertion on the size of the list be better?
https://www.baeldung.com/hamcrest-core-matchers

Reply:
Tried it, but i want to assert that items from brage entry keep original order, but has item + size does not care about order.